tools/iceoryx_build_test.sh

```diff
@@ -24,109 +24,139 @@ set -e
-            echo "    help                  Prints this help"
-            echo ""
-            echo "e.g. iceoryx_build_test.sh clean test release"
-            exit 0
-            ;;
-        *)
-            echo "Invalid argument '$arg'. Try 'help' for options."
-            exit -1
-            ;;
-    esac
+DDS_GATEWAY_FLAG="on"
+CYCLONEDDS_FLAG="$DDS_GATEWAY_FLAG"
+
+while (( "$#" )); do
+  case "$1" in
+    -b|--builddir)
+        BUILD_DIR=$(realpath $2)
+        shift 2
+        ;;
+    "clean")
+        CLEAN_BUILD=true
+        shift 1
+        ;;
+    "release")
+        BUILD_TYPE="Release"
+        shift 1
+        ;;
+    "debug")
+        BUILD_TYPE="Debug"
+        shift 1
+        ;;
+    "strict")
+        STRICT_FLAG="on"
+        shift 1
+        ;;
+    "test")
+        RUN_TEST=true
+        TEST_FLAG="on"
+        shift 1
+        ;;
+    "build-test")
+        RUN_TEST=false
+        TEST_FLAG="on"
+        shift 1
+        ;;
+    "skip-introspection")
+        INTROSPECTION_FLAG="off"
+        shift 1
+        ;;
+    "skip-dds-gateway")
+        DDS_GATEWAY_FLAG="on"
+        CYCLONEDDS_FLAG="on"
+        shift 1
+        ;;
+    "help")
+        echo "Build script for iceoryx."
+        echo "By default, iceoryx, the dds gateway and the examples are built."
+        echo ""
+        echo "Usage:"
+        echo "    iceoryx_build_test.sh [--builddir <dir>] [<args>]"
+        echo "Options:"
+        echo "    -b --builddir         Specify a non-default build directory"
+        echo "Args:"
+        echo "    clean                 Cleans the build directory"
+        echo "    release               Build release configuration"
+        echo "    debug                 Build debug configuration"
+        echo "    strict                Build is performed with '-Werror'"
+        echo "    test                  Builds and runs the tests"
+        echo "    build-test            Builds the tests (doesn't tun)"
+        echo "    skip-introspection    Skips building iceoryx introspection"
+        echo "    skip-dds-gateway      Skips building iceoryx dds gateway"
+        echo "    help                  Prints this help"
+        echo ""
+        echo "e.g. iceoryx_build_test.sh -b ./build-scripted clean test release"
+        exit 0
+        ;;
+    *)
+        echo "Invalid argument '$1'. Try 'help' for options."
+        exit -1
+        ;;
+  esac
 done
 
+# define directories dependent on the build directory
+ICEORYX_INSTALL_PREFIX=$BUILD_DIR/install/prefix/
+
+echo " [i] Building in $BUILD_DIR"
+
 #====================================================================================================
 #==== Step 1 : Build  ===============================================================================
 #====================================================================================================
 
-# Clean build folder
+# detect number of course if possible
+NUM_CORES=1
+if nproc >/dev/null 2>&1; then
```

Comment:
It would be better if the user can set the number of cores by himself. When i compile directly on Raspberry Pi it exhaust the complete ressources.
Maybe it can be autodetect by default but the user can also set the number of cores as argument.

Reply:
Makes sense, I will add the option.